server/src/main/java/org/elasticsearch/cluster/coordination/StableMasterService.java

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.coordination;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.ClusterChangedEvent;
+import org.elasticsearch.cluster.ClusterStateListener;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.health.HealthStatus;
```

Comment:
This creates a circular dependency - the `cluster.coordination` package depends on the `health`package and `health` depends on `cluster.coordination`.

I think we should either have a status entity in `cluster.coordination` and interpret it when this service is consumed, or don't have a status at all but a different way to express the result of this service - a diagnostic /analysis of sorts

Reply:
Oh good catch. I'll fix that.